src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3934,6 +3934,40 @@ static QVariant fcnRotate( const QVariantList &values, const QgsExpressionContex
   return QVariant::fromValue( fGeom );
 }
 
+static QVariant fcnAffineTransform( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry fGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+
+  double deltaX = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
+  double deltaY = QgsExpressionUtils::getDoubleValue( values.at( 2 ), parent );
+
+  double rotationZ = QgsExpressionUtils::getDoubleValue( values.at( 3 ), parent );
+
+  double scaleX = QgsExpressionUtils::getDoubleValue( values.at( 4 ), parent );
+  double scaleY = QgsExpressionUtils::getDoubleValue( values.at( 5 ), parent );
+
+  double deltaZ = QgsExpressionUtils::getDoubleValue( values.at( 6 ), parent );
+  double deltaM = QgsExpressionUtils::getDoubleValue( values.at( 7 ), parent );
+  double scaleZ = QgsExpressionUtils::getDoubleValue( values.at( 8 ), parent );
+  double scaleM = QgsExpressionUtils::getDoubleValue( values.at( 9 ), parent );
+
+  if ( deltaZ != 0.0 && !fGeom.constGet()->is3D() ) {
```

Comment:
This will crash if fGeom is null. Better check for that case immediately after retrieving the geometry argument and returning early if so.